R/getBaselineDivergence.R

```diff
@@ -330,13 +328,15 @@ setMethod("addBaselineDivergence", signature = c(x = "SummarizedExperiment"),
 
 # This function converts time divergence results to DF object
 .convert_divergence_to_df <- function(
-        x, res, time_res, name = "divergence", name.time = "time_diff", ...){
-    #
-    temp <- .check_input(name, list("character scalar"))
-    #
-    temp <- .check_input(name.time, list("character scalar"))
-    #
-    df <- DataFrame(res, time_res, row.names = colnames(x))
-    colnames(df) <- c(name, name.time)
+        x, res, reference, time_res, name = c("divergence", 
+                                              "time_diff", "ref_samples"), ...){
+    # Validate 'name' param
+    if (length(name) != 3) {
+        stop("'name' must be a vector of length 3: 
+             c('divergence', 'time_diff', 'ref_samples')", call. = FALSE)
+    }
+    temp <- .check_input(name[1:3], list("character vector"))
```

Comment:
To `.check_input()` you can add length parameter that can take the required length into account